src/main/java/bg/codexio/springframework/data/jpa/requery/adapter/GraphQLHttpFilterAdapter.java

```diff
@@ -0,0 +1,281 @@
+                            extractName(argument.getName()),
+                            extractedValue,
+                            getOperationFromArgument(argument.getName())
+                    );
+                    filterRequests.add(filterRequest);
+                }
+            }
+        }
+        return filterRequests;
+    }
+
+    private void handleMapValue(
+            String containingObjectName,
+            Object extractedValue,
+            List<FilterRequest> filterRequests
+    ) {
+        if (extractedValue instanceof Map) {
+            for (Map.Entry<String, Object> entry : ((Map<String, Object>) extractedValue).entrySet()) {
+                var nestedValue = (Value<?>) entry.getValue();
+                var nestedExtractedValue = extractValue(nestedValue);
+                if (nestedExtractedValue instanceof Map) {
+                    handleMapValue(entry.getKey(), nestedExtractedValue, filterRequests);
+                }
+
+                var filterRequest = new FilterRequest(
+                        containingObjectName.concat(".").concat(extractName(entry.getKey())),
+                        nestedExtractedValue,
+                        getOperationFromArgument(entry.getKey())
+                );
+                filterRequests.add(filterRequest);
+            }
+        }
+    }
+
+    private String extractName(String name) {
+        String[] suffixes = {
+                "_gt",
+                "_gte",
+                "_lt",
+                "_lte",
+                "_in",
+                "_not_in",
+                "_contains",
+                "_starts_with",
+                "_ends_with",
+                "_empty",
+                "_not_empty",
+                "_begins_with_caseins",
+                "_ends_with_caseins",
+                "_contains_caseins"
+        };
+
+        for (var suffix : suffixes) {
+            if (name.endsWith(suffix)) {
+                return name.substring(0, name.length() - suffix.length());
+            }
+        }
+
+        return name;
+    }
+
+    private Object extractValue(Value<?> value) {
+        if (value instanceof StringValue) {
+            return ((StringValue) value).getValue();
+        } else if (value instanceof IntValue) {
+            return ((IntValue) value).getValue();
+        } else if (value instanceof BooleanValue) {
+            return ((BooleanValue) value).isValue();
+        } else if (value instanceof FloatValue) {
+            return ((FloatValue) value).getValue();
+        } else if (value instanceof EnumValue) {
+            return ((EnumValue) value).getName();
+        } else if (value instanceof ArrayValue) {
+            return ((ArrayValue) value).getValues().stream().map(this::extractValue).toList();
+        } else if (value instanceof ObjectValue) {
+            return handleComplexObject((ObjectValue) value);
+        } else {
+            return null;
+        }
+    }
```

Comment:
Avoid such long if else chains as it will go through each condition. Switch is appropriate here. Something like:
```suggestion
    private Object extractValue(Value<?> value) {
        return switch (value) {
            case StringValue sv -> sv.getValue();
            case IntValue iv -> iv.getValue();
            case BooleanValue bv -> bv.isValue();
            case FloatValue fv -> fv.getValue();
            case EnumValue ev -> ev.getName();
            case ObjectValue ov -> handleComplexObject(ov);
            case ArrayValue av -> av.getValues()
                                    .stream()
                                    .map(this::extractValue)
                                    .toList();
            default -> null;
        };
    }
```

Reply:
adopted the change, thanks, done